middleman-core/lib/middleman-core/util/data.rb

```diff
@@ -57,6 +57,7 @@ def parse(full_path, known_type=nil)
   # @return [Hash]
   Contract String, Pathname, Bool => Hash
   def parse_yaml(content, full_path)
+    content.sub!(/\\---/, '---')
```

Comment:
Yes, I see my error now, thanks. I see now how I can modify the frontmatter in the parse method.